setup_wizard/character_rig_setup/rig_script.py

```diff
@@ -9,4 +14,1848 @@ def rig_character(
+    bpy.data.collections.remove(bpy.data.collections.get("append_Face Plate"),do_unlink=True)
+    bpy.data.collections.remove(bpy.data.collections.get("append_Eyes"),do_unlink=True)
+
+    # Adding Shape Key Drivers
+    ourRig = char_name+"Rig"
+
+    # Loop through skirt bones list. 
+
+    # converts deg to calc
+    def rad(num):
+        return num * (pi/180) 
+
+    # Perform more operations on number if needed.
+    def calc(num, cut=False):
+        percent = 0.8
+        if cut:
+            return rad(num*percent)
+        else: return rad(num)
+
+    # For front bones adjust number by depth
+    def add_const(skirt_bone, name, bone, expression, driver=True, trans_rot="ROT_X", f_min_x=calc(-1), f_max_x = calc(1), f_min_y = 0, f_max_y = 0, f_min_z = 0, f_max_z = 0, map_x='X', map_y='Y', map_z='Z', t_min_x=0, t_max_x=0, t_min_y=0, t_max_y=0, t_min_z=0, t_max_z=0):
+        armature = bpy.context.scene.objects[ourRig]
+
+        this_bone = bpy.context.scene.objects[char_name+"Rig"].pose.bones[skirt_bone]
+        co = this_bone.constraints.new('TRANSFORM')
+        co.name = name
+        co.target = our_char
+        co.subtarget = bone
+        co.use_motion_extrapolate = True
+
+        if driver:
+            influence_driver = co.driver_add("influence").driver
+            # DRIVER STUFF
+            var = influence_driver.variables.new()
+            var.name = "bone"
+            var.type = 'TRANSFORMS'
+            
+            var.targets[0].id = armature
+            var.targets[0].bone_target = bone
+            var.targets[0].transform_space = 'LOCAL_SPACE'
+            var.targets[0].transform_type = trans_rot
+
+            influence_driver.type = 'SCRIPTED'
+            influence_driver.expression = expression 
+
+            depsgraph = bpy.context.evaluated_depsgraph_get()
+            depsgraph.update()
+            # END DRIVER STUFF
+            
+        co.target_space = "LOCAL"
+        co.owner_space = "LOCAL"
+        co.map_from = "ROTATION"
+        co.map_to = "ROTATION"
+        
+        # Do transform constraint math
+        # MAP FROM (put in t/e block?)
+        co.from_min_x_rot = f_min_x
+        co.from_max_x_rot = f_max_x
+        
+        co.from_min_y_rot = f_min_y
+        co.from_max_y_rot = f_max_y
+        
+        co.from_min_z_rot = f_min_z
+        co.from_max_z_rot = f_max_z
+        
+        # MAP TO
+        co.map_to_x_from = map_x
+        co.to_min_x_rot = t_min_x
+        co.to_max_x_rot = t_max_x
+        
+        co.map_to_y_from = map_y
+        co.to_min_y_rot = t_min_y
+        co.to_max_y_rot = t_max_y
+        
+        co.map_to_z_from = map_z
+        co.to_min_z_rot = t_min_z
+        co.to_max_z_rot = t_max_z
+         
+    def leg_match(bone_name):
+        if " R " in bone_name:
+            return "DEF-thigh.R"
+        
+        elif " L " in bone_name:
+            return "DEF-thigh.L"
+
+    allow_back_move = True
```

Comment:
```suggestion

```

Moving the initial declaration from here to outside of the scope of the function.